tools/reactive-controllers/src/PendingState.ts

```diff
@@ -73,17 +73,27 @@ export class PendingStateController<T extends HostWithPendingState>
         const { pending, disabled, pendingLabel } = this.host;
         const currentAriaLabel = this.host.getAttribute('aria-label');
 
-        if (pending && !disabled && currentAriaLabel !== pendingLabel) {
-            // Cache the current `aria-label` to be restored when no longer `pending`
+        // If the current `aria-label` is different from the pending label, cache it
+        // or if the cached `aria-label` is different from the current `aria-label`, cache it
+        if (
+            (!this.cachedAriaLabel &&
```

Comment:
If this.cachedAriaLabel is undefined, isn't the second branch (this.cachedAriaLabel !== currentAriaLabel) always true? Also this is quite complex to read. Can you abstract these into a helper function, added a small snippet below if you want to take reference. Just making sure the main method is more readable
```ts
if (shouldCacheAriaLabel(this.cachedAriaLabel, currentAriaLabel, pendingLabel)) {
    this.cachedAriaLabel = currentAriaLabel;
}
...
function shouldCacheAriaLabel(cached, current, pending) {
    return (
        (!cached && current && current !== pending) ||
        (cached !== current && current && current !== pending)
    );
}
```

Reply:
the first half is checking if there is a cached label at all:
1. Save > Pending > Save

the second half is checking if the button label changes after pending that the cached label gets updated with the new label:
2. Save > Pending > Continue

This keeps everything in sync if the button goes to pending again. 

I like the suggestion for the helper function, i will add that in.